code/game/objects/items/reagent_containers/glass.dm

```diff
@@ -230,6 +231,93 @@
+	icon = 'icons/obj/items/tank.dmi'
+	icon_state = "mini_reagent_tank"
+	matter = list("metal" = 500)
+	attack_speed = 4
+	volume = 180
+	w_class = SIZE_MEDIUM
+	splashable = FALSE
+	can_be_placed_into = list(
+		/obj/structure/machinery/chem_master/,
+		/obj/structure/machinery/chem_dispenser/,
+		/obj/structure/machinery/reagentgrinder,
+		/obj/structure/surface/table,
+		/obj/structure/closet,
+		/obj/structure/sink,
+		/obj/item/storage,
+		/obj/item/clothing,
+		/obj/structure/machinery/bot/medbot,
+		/obj/structure/machinery/computer/pandemic,
+		/obj/item/storage/secure/safe,
+		/obj/structure/machinery/iv_drip,
+		/obj/structure/machinery/disposal,
+		/obj/structure/machinery/sleeper,
+		/obj/structure/machinery/smartfridge/,
+		/obj/structure/machinery/biogenerator,
+		/obj/structure/machinery/reagent_analyzer,
+		/obj/structure/machinery/centrifuge,
+		/obj/structure/machinery/autodispenser,
+		/obj/structure/machinery/constructable_frame)
+
+/obj/item/reagent_container/glass/minitank/on_reagent_change()
+	update_icon()
+
+
+/obj/item/reagent_container/glass/minitank/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/reagent_container/hypospray/autoinjector))
+		var/obj/item/reagent_container/hypospray/autoinjector/A = W
+		if(A.mixed_chem)
+			to_chat(user, SPAN_WARNING("The autoinjector doesn't fit into the [src]'s valve. It's probably not compatible."))
+			return
+		if(reagents.has_reagent(A.chemname, A.volume))
+			reagents.trans_id_to(A, A.chemname, A.volume)
+			A.uses_left = 3
+			A.update_icon()
+			playsound(src.loc, 'sound/effects/refill.ogg', 25, 1, 3)
+		else
+			to_chat(user, SPAN_WARNING("A small LED on the [src] blinks. The tank is either empty or not compatible with that!"))
+			. = ..()
+			return
+		to_chat(user,SPAN_INFO("You successfully refill [W.name] with the [src]!"))
+
+/obj/item/reagent_container/glass/minitank/verb/flush_tank(mob/user)
+	set category = "Object"
+	set name = "Flush tank"
+	set src in view(1)
+	playsound(src.loc, 'sound/effects/refill.ogg', 25, 1, 3)
+	to_chat(user, SPAN_WARNING("You work the flush valve and successfully flush the [src]!"))
+	if(!usr || usr.stat || usr.lying)	return
+	reagents.clear_reagents()
+	return
+/obj/item/reagent_container/glass/minitank/attack_self()
+	..()
+	return
+
+/obj/item/reagent_container/glass/minitank/update_icon()
+	overlays.Cut()
+
+	if(reagents && reagents.total_volume)
+		var/image/filling = image('icons/obj/items/reagentfillings.dmi', src, "[icon_state]10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(0 to 24) 		filling.icon_state = "[icon_state]10"
+			if(25 to 49)		filling.icon_state = "[icon_state]25"
+			if(50 to 74)		filling.icon_state = "[icon_state]50"
+			if(75 to 84)		filling.icon_state = "[icon_state]75"
+			if(85 to INFINITY)	filling.icon_state = "[icon_state]100"
```

Comment:
you can do this in a neater way by making the number after the icon state a var and having that var round to the nearest 25 off var/percent

Reply:
care to elaborate more? I'm confused.